File: app/src/main/java/ru/neosvet/vestnewage/fragment/WelcomeFragment.java
package ru.neosvet.vestnewage.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import ru.neosvet.utils.Const;
import ru.neosvet.vestnewage.R;
import ru.neosvet.vestnewage.list.ListAdapter;
import ru.neosvet.vestnewage.list.ListItem;

public class WelcomeFragment extends BottomSheetDialogFragment {
    public interface ItemClicker {
        void onItemClick(String link);
    }

    private ItemClicker clicker;
    private ListAdapter adapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.welcome_fragment, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        getDialog().setOnShowListener(dialog -> {
            BottomSheetDialog d = (BottomSheetDialog) dialog;
            FrameLayout bottomSheet = d.findViewById(com.google.android.material.R.id.design_bottom_sheet);
            BottomSheetBehavior.from(bottomSheet).setPeekHeight(
                    (int) (175 * getResources().getDisplayMetrics().density));
        });

        ListView lvBottom = view.findViewById(R.id.lvBottom);
        adapter = new ListAdapter(requireContext());
        lvBottom.setAdapter(adapter);
        lvBottom.setOnItemClickListener((adapterView, view1, pos, l) -> {
            if (clicker != null)
                clicker.onItemClick(adapter.getItem(pos).getLink());
        });
        fillInList();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof ItemClicker)
            clicker = (ItemClicker) context;
    }

    private void fillInList() {
        Bundle args = getArguments();
        if (args == null)
            return;

        if (args.getBoolean(Const.ADS)) {
            adapter.addItem(new ListItem(getString(R.string.new_dev_ads), Const.ADS));
        }

        if (args.getBoolean(Const.PAGE)) {
            String[] title = args.getStringArray(Const.TITLE);
            String[] link = args.getStringArray(Const.LINK);
            for (int i = 0; i < title.length; i++) {
                ListItem item = new ListItem(title[i], link[i]);
                item.setDes(getString(R.string.updated_page));
                adapter.addItem(item);
            }
        }

        if (args.getBoolean(Const.TIME)) {
            int timeDiff = args.getInt(Const.TIMEDIFF);
            ListItem item = new ListItem(getString(R.string.sync_time));
            if (timeDiff == 0)
                item.setDes(getString(R.string.matches));
            else
                item.setDes(String.format(getString(R.string.time_deviation),
                        timeDiff));
            adapter.addItem(item);
            adapter.notifyDataSetChanged();
        }
    }
}

File: app/src/main/java/ru/neosvet/vestnewage/view/fragment/WelcomeFragment.kt
package ru.neosvet.vestnewage.view.fragment

import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import ru.neosvet.vestnewage.R
import ru.neosvet.vestnewage.data.BasicItem
import ru.neosvet.vestnewage.utils.Const
import ru.neosvet.vestnewage.utils.ScreenUtils
import ru.neosvet.vestnewage.view.basic.convertDpi
import ru.neosvet.vestnewage.view.list.RecyclerAdapter

class WelcomeFragment : BottomSheetDialogFragment() {
    interface ItemClicker {
        fun onItemClick(link: String)
    }

    companion object {
        fun newInstance(hasNew: Boolean, timediff: Int): WelcomeFragment {
            val fragment = WelcomeFragment()
            val args = Bundle()
            args.putBoolean(Const.ADS, hasNew)
            args.putInt(Const.TIMEDIFF, timediff)
            fragment.arguments = args
            return fragment
        }
    }

    private var clicker: ItemClicker? = null
    val list = mutableListOf<BasicItem>()
    private val adapter = RecyclerAdapter(this::onItemClick)

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.welcome_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        dialog!!.setOnShowListener { dialog: DialogInterface ->
            val d = dialog as BottomSheetDialog
            val sheetId = com.google.android.material.R.id.design_bottom_sheet
            d.findViewById<View>(sheetId)?.let { bottomSheet ->
                val height = requireContext().convertDpi(175)
                BottomSheetBehavior.from(bottomSheet).setPeekHeight(height)
            }
        }
        val rvBottom = view.findViewById(R.id.rvBottom) as RecyclerView
        rvBottom.layoutManager = GridLayoutManager(requireContext(), ScreenUtils.span)
        rvBottom.adapter = adapter
        arguments?.let { parseArguments(it) }
        adapter.setItems(list)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is ItemClicker)
            clicker = context
    }

    private fun parseArguments(args: Bundle) {
        val timeDiff = args.getInt(Const.TIMEDIFF)
        val item = BasicItem(getString(R.string.sync_time))
        item.des = if (timeDiff == 0)
            getString(R.string.matches)
        else
            String.format(getString(R.string.time_deviation), timeDiff)
        list.add(0, item)
        if (args.getBoolean(Const.ADS))
            list.add(0, BasicItem(getString(R.string.new_dev_ads), Const.ADS))
    }

    private fun onItemClick(index: Int, item: BasicItem) {
        clicker?.onItemClick(item.link)
        if (item.link == Const.ADS) dismiss()
    }
}